sotodlib/site_pipeline/update_preprocess_plots.py

```diff
@@ -0,0 +1,256 @@
+            groups = context.obsfiledb.get_detsets(obs_id)
+            return group_by, [[g] for g in groups]
+        
+    det_info = context.get_det_info(obs_id)
+    rs = det_info.subset(keys=group_by).distinct()
+    groups = [[b for a,b in r.items()] for r in rs]
+    return group_by, groups
+
+def plot_preprocess_tod(obs_id, configs, context, group_list=None, verbosity=2):
+    """ Loads the saved information from the preprocessing pipeline and runs the
+    plotting section of the pipeline. 
+
+    Assumes preprocess_tod has already been run on the requested observation. 
+    
+    Arguments
+    ----------
+    obs_id: multiple
+        passed to `context.get_obs` to load AxisManager, see Notes for 
+        `context.get_obs`
+    configs: string or dictionary
+        config file or loaded config directory
+    """
+    logger = sp_util.init_logger("preprocess", verbosity=verbosity)
+    
+    group_by, groups = _get_groups(obs_id, configs, context)
+    all_groups = groups.copy()
+    for g in all_groups:
+        if group_list is not None:
+            if g not in group_list:
+                groups.remove(g)
+                continue
+        if 'wafer.bandpass' in group_by:
+            if 'NC' in g:
+                groups.remove(g)
+                continue
+        try:
+            meta = context.get_meta(obs_id, dets = {gb:gg for gb, gg in zip(group_by, g)})
+        except Exception as e:
+            errmsg = f'{type(e)}: {e}'
+            tb = ''.join(traceback.format_tb(e.__traceback__))
+            logger.info(f"ERROR: {obs_id} {g}\n{errmsg}\n{tb}")
+            groups.remove(g)
+            continue
+
+        if meta.dets.count == 0:
+            groups.remove(g)
+        
+    if len(groups) == 0:
+        logger.warning(f"group_list:{group_list} contains no overlap with "
+                       f"groups in observation: {obs_id}:{all_groups}. "
+                       f"No analysis to run.")
+        return
+    
+    pipe = Pipeline(configs["process_pipe"], plot_dir=configs["plot_dir"], logger=logger)
+    
+    for group in groups:
+        logger.info(f"Beginning run for {obs_id}:{group}")
+        try:
+            meta = context.get_meta(obs_id, dets={gb:g for gb, g in zip(group_by, group)})
+            aman = context.get_obs(meta)
+            pipe.run(aman, aman.preprocess, update_plot=True)
+        except Exception as e:
+            errmsg = f'{type(e)}: {e}'
+            tb = ''.join(traceback.format_tb(e.__traceback__))
+            logger.info(f"ERROR: {obs_id} {group}\n{errmsg}\n{tb}")
+            continue
+
+
+def get_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+    parser.add_argument('configs', help="Preprocessing Configuration File")
+    parser.add_argument(
+        '--query', 
+        help="Query to pass to the observation list. Use \\'string\\' to "
+             "pass in strings within the query.",  
+        type=str
+    )
```

Comment:
Seems like if this is purely for updating from an existing database, rather than passing a query you just want to pass the database and get back the list of obs-ids and groups from there. This would change this set of arguments a lot.

Reply:
I can imagine cases where a custom-formatted ManifestDB would be less general to parse, so I figured using the more standard query to get the obs and groups from the same preprocess config that would have been used initially, containing the path to the ManifestDB, would be the most straightforward method to avoid any possible differences.

Reply:
Latest commits add this capability by setting `use_db=True` (aka `--use-db` via command line). The obs list and groups are obtained directly from the database instead of needing a query.